src/main/java/cn/nukkit/block/BlockGlowLichen.java

```diff
@@ -0,0 +1,269 @@
+    }
+
+    @Since("FUTURE")
+    @PowerNukkitOnly
+    @Nonnull
+    @Override
+    public BlockProperties getProperties() {
+        return PROPERTIES;
+    }
+
+    public BlockFace[] getGrowthSides() {
+        Stream<BlockFace>returns = Arrays.stream(BlockFace.values()).filter(this::isGrowthToSide);
+        return returns.toArray(BlockFace[]::new);
+    }
+
+    public boolean isGrowthToSide(@Nonnull BlockFace side) {
+        return ((getPropertyValue(MULTI_FACE_DIRECTION_BITS) >> side.getIndex()) & 0x1) > 0;
+    }
+
+    public void growToSide(BlockFace side) {
+        if (!isGrowthToSide(side)) {
+            setPropertyValue(MULTI_FACE_DIRECTION_BITS, getPropertyValue(MULTI_FACE_DIRECTION_BITS) | (0b0001 << side.getIndex()));
+            getLevel().setBlock(this, this, true, true);
+        }
+    }
+
+    @Override
+    public boolean place(@Nonnull Item item, @Nonnull Block block, @Nonnull Block target, @Nonnull BlockFace face, double fx, double fy, double fz, @Nullable Player player) {
+
+        if (!target.isSolid() && target.getId() != BlockID.GLOW_LICHEN) {
+            return false;
+        }
+
+        int currentMeta = 0;
+        if (block.getId() == GLOW_LICHEN) {
+            currentMeta = block.getPropertyValue(MULTI_FACE_DIRECTION_BITS);
+        }
+
+        setPropertyValue(MULTI_FACE_DIRECTION_BITS, currentMeta | (0b0001 << face.getOpposite().getIndex()));
+
+        if (getPropertyValue(MULTI_FACE_DIRECTION_BITS) == currentMeta) {
+            BlockFace[] sides = BlockFace.values();
+            Stream<BlockFace> faceStream = Arrays.stream(sides).filter(side ->
+                    block.getSide(side).isSolid() && !isGrowthToSide(side)
+            );
+            Optional<BlockFace> optionalFace = faceStream.findFirst();
+            if (optionalFace.isPresent()) {
+                growToSide(optionalFace.get());
+                return true;
+            }
+
+            return false;
+        }
+
+        getLevel().setBlock(block, this, true, true);
+        return true;
+    }
+
+    @Override
+    public boolean onActivate(@Nonnull Item item, @Nullable Player player) {
+
+        if (!item.isFertilizer()) {
+            return false;
+        }
+
+        Map<Block, BlockFace> candidates = new HashMap<>();
+
+        for (int i = 0; i < 5; i++) {
+            BlockFace side = BlockFace.fromIndex(i);
+            Block support = this.getSide(side);
+
+            if (isGrowthToSide(side)) {
+                BlockFace[] supportSides = side.getEdges().toArray(new BlockFace[0]);
+
+                for (BlockFace supportSide : supportSides) {
+                    Block supportNeighbor = support.getSide(supportSide);
+
+                    if (supportNeighbor.getId() == BlockID.AIR) {
+                        candidates.put(supportNeighbor, supportSide.getOpposite());
+                    }
+
+                    if (!supportNeighbor.isSolid()) {
+                        continue;
+                    }
```

Comment:
This condition could be the first one since its exclusive with the first one and eliminates checking for a solid block twice.